sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveUDFSuite.scala

```diff
@@ -141,11 +141,13 @@ class HiveUDFSuite extends QueryTest with TestHiveSingleton with SQLTestUtils {
   }
 
   test("SPARK-2693 udaf aggregates test") {
-    checkAnswer(sql("SELECT percentile(key, 1) FROM src LIMIT 1"),
-      sql("SELECT max(key) FROM src").collect().toSeq)
+    checkAnswer(
+      sql("SELECT percentile(key, 1) FROM src LIMIT 1"),
+      sql("SELECT double(max(key)) FROM src"))
```

Comment:
`percentile` always return  double, we need to cast `max` to double so that we can compare the results.